Migrate dashboard to grafanalib

francoisferrand opened the pull request:
Issue: CLDSRV-147

francoisferrand committed:
Migrate dashboard to grafanalib

Issue: CLDSRV-147

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

lavigne958 commented on monitoring/dashboard.py:
```diff
@@ -0,0 +1,412 @@
+    )],
+    thresholds=[
+        Threshold("green", 0, 0.0),
+        Threshold("red",   1, 80.0),
```
Why 80  as a "red" value ?

francoisferrand replied:
this is same as Vault, just an arbitrary limit I guess
It should generally stay low since it counts "in flight" requests only; but maybe 80 is normal in a loaded system.
Since it just changes the color of the panel, I guess this should be fine anyway : what do you think?

francoisferrand committed:
Make flake8 compliant

Issue: CLDSRV-147

lavigne958 commented on monitoring/dashboard.py:
```diff
@@ -0,0 +1,412 @@
+            'sum(rate(http_request_duration_seconds_count{namespace="${namespace}", job="${job}"' + extra + "}[$__rate_interval]))",
+        ]),
+        legendFormat=title,
+    )
```
a new line here could help code readability

lavigne958 commented on monitoring/dashboard.py:
```diff
@@ -0,0 +1,412 @@
+            "id": "custom.axisPlacement", 
+            "value": mode,
+        }],
+    }
```
same here a new line could help to read code

francoisferrand committed:
Remove type annotations

The version of flake8 in Eve's worker does not seem to support
type annotations, and fails with: `E999 SyntaxError: invalid syntax`.

Issue: CLDSRV-147

alexanderchan-scality commented on monitoring/dashboard.py:
```diff
@@ -0,0 +1,452 @@
+            ConstantInput(
+                name="job",
+                label="job",
+                description="Name of the Vault job, used to filter the Vault "
```
the description mentions vault. is this correct?

francoisferrand replied:
good catch, thanks!

francoisferrand commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.4`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.2`
* `development/8.3`
* `stabilization/7.10.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.4`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.2`
* `development/8.3`
* `stabilization/7.10.3`



Please check the status of the associated issue CLDSRV-147.

Goodbye francoisferrand.
